src/azure-cli/azure/cli/command_modules/role/_help.py

```diff
@@ -115,6 +115,11 @@
     The output includes credentials that you must protect. Be sure that you do not include these credentials
     in your code or check the credentials into your source control. As an alternative, consider using
     [managed identities](https://aka.ms/azadsp-managed-identities) if available to avoid the need to use credentials.
+    
+    **Note:** This command resets the credentials of the Service Principal, not the App Registration.
```

Comment:
This is a good catch! But I guess you put this statement at the wrong place? `ad app credential reset` does reset the credential of the application, not service principal.

The help message for `az ad sp credential reset` is here:

https://github.com/Azure/azure-cli/blob/33cbd01f75c7d4c3814de5a1cbf6f42de7890467/src/azure-cli/azure/cli/command_modules/role/_help.py#L527-L531

As for the behavior change, we called out in https://docs.microsoft.com/en-us/cli/azure/microsoft-graph-migration#az-ad-sp-credential